src/Relation/Nullary/Negation/Core.agda

```diff
@@ -48,23 +47,26 @@ _¬-⊎_ = [_,_]
 ------------------------------------------------------------------------
 -- Uses of negation
 
+weak-contradiction : .A → ¬ A → Whatever
+weak-contradiction a ¬a = ⊥-elim-irr (¬a a)
+
 contradiction : A → ¬ A → Whatever
-contradiction a ¬a = ⊥-elim (¬a a)
+contradiction a = weak-contradiction a
 
 contradiction₂ : A ⊎ B → ¬ A → ¬ B → Whatever
-contradiction₂ (inj₁ a) ¬a ¬b = contradiction a ¬a
-contradiction₂ (inj₂ b) ¬a ¬b = contradiction b ¬b
+contradiction₂ (inj₁ a) ¬a ¬b = weak-contradiction a ¬a
+contradiction₂ (inj₂ b) ¬a ¬b = weak-contradiction b ¬b
```

Comment:
Are these refactorings really necessary? It's not like they rely on the contradiction being weak? I would advocate for keeping them using `contradiction` except where necessary? Not to mention it's just plain shorter.

Reply:
Have reverted.